lms/static/js/spec/discovery/discovery_spec.js

```diff
@@ -259,51 +259,135 @@ define([
 
     });
 
-
-    describe('FilterView', function () {
-
-        beforeEach(function () {
-            TemplateHelpers.installTemplate('templates/discovery/filter');
-            this.filter = new FilterView({
-                model: new FilterModel()
-            });
-        });
-
-        it('model cleans view on destruction correctly', function () {
-            var data = this.filter.model.attributes;
-            this.filter.render();
-            expect(this.filter.$el.length).toBe(1);
-            expect(this.filter.$el).toContainHtml(data.content.query);
-            this.filter.cleanModelView();
-            expect(this.filter.$el.length).toBe(0);
-        });
-
-    });
-
-
     describe('FilterBarView', function () {
-
         beforeEach(function () {
-            TemplateHelpers.installTemplate('templates/discovery/filter_bar_view');
+            loadFixtures('js/fixtures/discovery.html');
+            TemplateHelpers.installTemplates(
+                ['templates/discovery/filter_bar',
+                'templates/discovery/filter']
+            );
             this.filterBar = new FiltersBarView();
+            this.onClear = jasmine.createSpy('onClear');
+            this.filterBar.on('clear', this.onClear);
         });
 
         it('view searches for sent facet object', function () {
             expect(this.filterBar.$el.length).toBe(1);
-            this.filter.addFilter(FACET_LIST[0]);
+            this.filterBar.addFilter(FACET_LIST[0]);
             expect(this.filterBar.$el.find('#clear-all-filters')).toBeVisible();
         });
 
         it('view searches for entered search string', function () {
             spyOn(this.filterBar, 'addFilter').andCallThrough();
             expect(this.filterBar.$el.length).toBe(1);
-            this.filter.changeQueryFilter(SEARCH_FILTER.query);
+            this.filterBar.changeQueryFilter(SEARCH_FILTER.query);
             expect(this.filterBar.$el.find('#clear-all-filters')).toBeVisible();
             expect(this.filterBar.addFilter).toHaveBeenCalledWith(SEARCH_FILTER);
         });
 
+        it('model cleans view on destruction correctly', function () {
+            this.filterBar.addFilter(SEARCH_FILTER);
+            var model = this.filterBar.collection.findWhere(SEARCH_FILTER);
+            expect(this.filterBar.$el.find('.active-filter').length).toBe(1);
+            model.cleanModelView();
+            expect(this.filterBar.$el.find('.active-filter').length).toBe(0);
+        });
+
+        it('view removes all filters and hiddes bar if clear all', function () {
+            spyOn(this.filterBar, 'clearAll').andCallThrough();
```

Comment:
This spy seems excessive, since this is covered by testing `this.onClear`. If I am not mistaken, it also breaks attached DOM events and makes it necessary to call `this.filterBar.delegateEvents();`.